meson.build

```diff
@@ -16,6 +16,8 @@ endif
 
 subdir('ctlra')
 
+#['-Wl,--whole-archive'] + devices_lib + ['-Wl,--no-whole-archive']
```

Comment:
Commented out code?

Reply:
Yep.. the new build options allows us to build shared / static libraries. That's good - except that with the refactoring of the registration of devices, the GCC __attribute__(constructor) trick doesn't work unless the static library `.a` is linked in.

In terms of the build system, it needs to act differently based on shared vs static builds. Shared works fine - on opening the `.so` the constructor registers the device, done. With static builds, it needs to statically link in "whole archive", and "no-as-needed", because the functions aren't actually called from the `libctlra.so`, so unless explicitly told to include the `.a` libs for each device, it'll just leave them out... and leaving them out means no devices are supported.

All in all, I played with it for a while to try make it work, but I didn't get it working yet..